src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,123 @@
+package storage;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+/**
+ * Handle the read and storage of data.
+ */
+public class Storage {
+    private String pathName;
+
+    public Storage(String pathName) {
+        this.pathName = pathName;
+    }
+
+    /**
+     * Convert the time from string to a LocalDate datatype.
+     * 
+     * @param time
+     * @return a LocalDate saving the date information.
+     */
+    public LocalDate readAsDate(String time) {
+        String[] parts = time.split(" ");
+        String datePart = parts[0];
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        LocalDate localDate = LocalDate.parse(datePart, formatter);
+        return localDate;
+    }
+
+    public void autoUpdate(ArrayList<Task> todoList) {
+        File file = new File(this.pathName);
+        file.getParentFile().mkdirs();
+        try {
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            BufferedWriter writer = new BufferedWriter(new FileWriter(file));
+            for (Task t : todoList) {
+                if (t instanceof Todo) {
```

Comment:
Avoid using instanceof, instead try to use polymorphism to achieve the same functionality

Reply:
Could you explain a bit more about how to use polymorphism to achieve it? Thanks